src/test/ui/error-trait/error-in-panic.run.stderr

```diff
@@ -0,0 +1,6 @@
+thread 'main' panicked at 'here's my panic error message', $DIR/error-in-panic.rs:49:5
+Source: my source error message
+
+Caused by:
+      my source's source error message
+note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Comment:
The format here is intentionally not intended to be the final format, right now it's fairly inconsistent. I'm not really sure exactly how we'd want to format this. My default suggestion would be something like 

```
thread 'main' panicked at 'here's my panic error message', $DIR/error-in-panic.rs:49:5

source: my source error message

caused by:
      my source's source error message

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace